File: src/main/java/me/creepinson/creepinoutils/util/network/IBaseTile.java
package me.creepinson.creepinoutils.util.network;

import dev.throwouterror.util.math.Tensor;
import me.creepinson.creepinoutils.util.IConnectable;
import me.creepinson.creepinoutils.util.upgrade.IUpgradeable;
import net.minecraft.tileentity.ITickableTileEntity;
import net.minecraft.util.math.BlockPos;

/**
 * @author Theo Paris https://theoparis.com
 **/
public interface IBaseTile extends IConnectable, IUpgradeable, ITickableTileEntity {
    boolean isActive();

    void refresh();

    void onNeighborChange(Tensor pos);

}

File: src/main/kotlin/com/theoparis/creepinoutils/util/api/IBaseTile.kt
package com.theoparis.creepinoutils.util.api

import com.theoparis.creepinoutils.util.upgrade.IUpgradeable
import net.minecraft.tileentity.ITickableTileEntity
import net.minecraft.util.math.BlockPos

interface IBaseTile : IConnectable, IUpgradeable, ITickableTileEntity {
    val isActive: Boolean
    fun refresh()
    fun onNeighborChange(pos: BlockPos?)
}